central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -245,10 +279,87 @@ func (m *managerImpl) ProcessRescanRequest(_ context.Context, _ interface{}) err
 
 // DeleteScan processes a request to delete an existing compliance scan configuration.
 // TODO(ROX-19540)
-func (m *managerImpl) DeleteScan(_ context.Context, _ interface{}) error {
+func (m *managerImpl) DeleteScan(ctx context.Context, scanID string) error {
 	// TODO:
 	// 1. Validate config exists in database
 	// 2. Lock config so it cannot be edited/updated
 	// 3. Push request to Sensor
-	panic("implement me")
+
+	// Check if scan configuration already exists.
+	scanConfig, found, err := m.scanSettingDS.GetScanConfiguration(ctx, scanID)
+	if err != nil {
+		log.Error(err)
+		return errors.Wrapf(err, "Unable to delete scan configuration ID %q.", scanID)
+	}
```

Comment:
My suggestion is that we completely re-order this file and push a lot of this functionality into the datastore. Inside the datastore, we can utilize the lock there. The flow could be like:

1. Delete if exists and do all datastore things
2. Push to sensors if successful

Also do we need to track the delete request? If the push to sensor fails then that would indicate the connection is gone, if the connection is gone sensor will need to completely resync the state in Central anyways. If we need to be wary of modifications and deletes happening concurrently then we can get creative

Reply:
Yeah I think we can simplify this a little bit.  For the creation of a config we needed to track the response to basically record what clusters had the config.  For the delete case we don't need any of that.  We will essentially need to delete the config from the scan configuration table and from the scan config status table that tells us which configurations are on which clusters.  So I concur with the suggestion above.  

I don't think we have to worry too much about deletes being concurrent at least not at the moment.  The unique constraint on the table provides us some freedom here as their can't be a new addition until the delete completes.

Reply:
thanks for all the reviews here, I think this makes more sense now, I have changed how to handle the deletion according to this, but I still need to update the mock unit test for them.